saleor/graphql/product/mutations/products.py

```diff
@@ -866,6 +866,21 @@ def clean_input(
                 cleaned_input["product"]
             )
 
+        variant_attributes_ids = [
+            graphene.Node.to_global_id("Attribute", attr_id)
+            for attr_id in list(
+                product_type.variant_attributes.all().values_list("pk", flat=True)
+            )
+        ]
+
+        # Check if given attributes are variant attributes
+        for attr in data["attributes"]:
+            if attr["id"] not in variant_attributes_ids:
+                raise ValidationError(
+                    "Given attribute is not a variant attribute",
+                    ProductErrorCode.ATTRIBUTE_CANNOT_BE_ASSIGNED.value,
+                )
```

Comment:
IMO we can simplify this by creating sets for both lists and make difference. Also, I will suggest returning the invalid attributes as `params` in error like:
```
raise ValidationError(
      "Given attribute is not a variant attribute.",
      code=ProductErrorCode.ATTRIBUTE_CANNOT_BE_ASSIGNED.value,
      params={"attributes": invalid_attributes}
)
```
👮‍♀️ Remember about dot at the end of the error message.